FWCore/Utilities/interface/propagate_const_array.h

```diff
@@ -19,18 +19,9 @@
 
 namespace edm {
 
+  template <typename T>
+  concept ArrayAddressable = requires(T& a) { a[0]; } or std::is_array_v<T>;
```

Comment:
Do we want this `ArrayAddressable` to be potentially used elsewhere, or is it more of an implementation detail of `propagate_const_array`?

Reply:
I was of two minds for that. I THINK it should only be here. If it proves useful, we could later move it.

Reply:
If we treat it as an implementation detail, should it be placed in `edm::impl` namespace?

Reply:
I thought about that but really, `edm::` and `edm::impl` are really not that much different for _polluting namespace_ reasons. If we REALLY wanted to be better than we should use something like `edm::propagate_impl` to better isolate names. 

I'm not really arguing against using `edm::impl`, it's just that we use that a lot.

Reply:
I was thinking more from the "hiding from users" point of view than (along `edm::impl` _hints_ that users shouldn't use it). I'd be perfectly fine with `edm::propagate_impl` here (and I agree with the name collision prevention argument).